kani-driver/src/cbmc_property_renderer.rs

```diff
@@ -0,0 +1,643 @@
+            ("pointer arithmetic: pointer outside object bounds", None),
+            ("pointer arithmetic: invalid integer address", None),
+        ],
+    );
+    map.insert(
+        "pointer_dereference",
+        vec![
+            (
+                "dereferenced function pointer must be",
+                Some("dereference failure: invalid function pointer"),
+            ),
+            ("dereference failure: pointer NULL", None),
+            ("dereference failure: pointer invalid", None),
+            ("dereference failure: deallocated dynamic object", None),
+            ("dereference failure: dead object", None),
+            ("dereference failure: pointer outside dynamic object bounds", None),
+            ("dereference failure: pointer outside object bounds", None),
+            ("dereference failure: invalid integer address", None),
+        ],
+    );
+    // These are very hard to understand without more context.
+    map.insert(
+        "pointer_primitives",
+        vec![
+            ("pointer invalid", None),
+            ("deallocated dynamic object", Some("pointer to deallocated dynamic object")),
+            ("dead object", Some("pointer to dead object")),
+            ("pointer outside dynamic object bounds", None),
+            ("pointer outside object bounds", None),
+            ("invalid integer address", None),
+        ],
+    );
+    map.insert(
+        "array_bounds",
+        vec![
+            ("lower bound", Some("index out of bounds")),
+            // This one is redundant:
+            // ("dynamic object upper bound", Some("access out of bounds")),
+            (
+                "upper bound",
+                Some("index out of bounds: the length is less than or equal to the given index"),
+            ),
+        ],
+    );
+    map.insert(
+        "bit_count",
+        vec![
+            ("count trailing zeros is undefined for value zero", None),
+            ("count leading zeros is undefined for value zero", None),
+        ],
+    );
+    map.insert("memory-leak", vec![("dynamically allocated memory never freed", None)]);
+    // These pre-conditions should not print temporary variables since they are embedded in the libc implementation.
+    // They are added via `__CPROVER_precondition`.
+    // map.insert("precondition_instance": vec![]);
+    map
+});
+
+const UNSUPPORTED_CONSTRUCT_DESC: &str = "is not currently supported by Kani";
+const UNWINDING_ASSERT_DESC: &str = "unwinding assertion loop";
+const DEFAULT_ASSERTION: &str = "assertion";
+const REACH_CHECK_DESC: &str = "[KANI_REACHABILITY_CHECK]";
+
+impl ParserItem {
+    /// Determines if an item must be skipped or not.
+    fn must_be_skipped(&self) -> bool {
+        matches!(&self, ParserItem::Message { message_text, .. } if message_text.starts_with("Building error trace") || message_text.starts_with("VERIFICATION"))
+    }
+}
+
+pub fn kani_property_filter(
```

Comment:
This name is not very descriptive. It's filtering more than just properties, right?